### PR TITLE
[gh-2417] add peel functions for bcs.move.

### LINKED ISSUE #2417: [moveos_std] decode bcs in Move
Introduce peel API for `bcs.move` module and enable decode bcs in Move.
 
https://github.com/MystenLabs/sui/blob/2887eea4a9438751da8ccb48bcb49a4e81673c0a/crates/sui-framework/packages/sui-framework/sources/bcs.move

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

Add `peel` function referred from Sui for `bcs.move`.

- Closes #2417

### CODE REVIEW

File: frameworks/moveos-stdlib/sources/bcs.move [hunk 1]
```diff
@@ -41,6 +46,327 @@ module moveos_std::bcs{
...
+        assert!(vector::length(&v) >= 32, ErrorInvalidLength);
+        let i = 0;
+        let addr_bytes = vector::empty<u8>();
+        while (i < 32) {
+            let byte = vector::pop_back(&mut v);
+            vector::push_back(&mut addr_bytes, byte);
+            i = i + 1;
+        };
+        to_address(addr_bytes)
+    }
```
Comment (jolestar) [inline frameworks/moveos-stdlib/sources/bcs.move:60]: We should define a `BCS` struct and use the struct as the argument.

**Review by jolestar — COMMENTED** [review]
Need testcases for peel functions.

File: frameworks/moveos-stdlib/sources/bcs.move (hunk 1 above)
Comment (feliciss) [reply frameworks/moveos-stdlib/sources/bcs.move:60]: > We should define a `BCS` struct and use the struct as the argument.

Okay.

**Comment by feliciss** [comment]
> Need testcases for peel functions.

I've updated test cases.

**Review by jolestar — APPROVED** [review]

**Pull request merged**